src/models/types/circuit/circCompDot.mc

```diff
@@ -0,0 +1,108 @@
+                    [
+                        ("black",side_width,side_height),
+                    ("none",center_width,side_height),
+                    ("none",(addi side_width 1),side_height)],
+            "</tr> 
+            <tr>",
+                (foldl (lam str. lam x. concat str (makeTDElem x.0 x.1 x.2 quote))) "" 
+                    [("black",side_width,side_height),("none",center_width,center_height),("black",(addi side_width 1),side_height)],
+            "</tr>
+            <tr>",
+                (foldl (lam str. lam x. concat str (makeTDElem x.0 x.1 x.2 quote))) "" 
+                    [("black",side_width,side_height), ("none",center_width,side_height),("none",(addi side_width 1),side_height)],
+            "</tr>   
+        </table>>"
+    ] in (setting,"V") in
+    formatComponentToDot name quote (concatList ["xlabel=",quote,value,settings.1,quote," ",settings.0]) isConnected
+
+-- returns the ground component in dot.
+let circGroundToDot = lam quote. lam name. lam custom_settings. lam isConnected.
+    let width =5 in
+    let height = 1 in
+    let settings = match custom_settings with Some (setting,unit) then (setting,unit) else
+        (foldl concat [] ["shape=none, color=none height=0 width=0 margin=0 label=<
+    <table CELLBORDER=",quote,"0",quote," CELLSPACING=",quote,"0",quote," CELLPADDING=",quote,"0",quote," >\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["black","black","black","black","black"],
+        " </tr>\n<tr>",
+           makeTDElem "none" width (muli 2 height) quote,
+       "</tr>\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["none","black","black","black","none"],
+        "</tr>\n<tr>",
+            makeTDElem "none" width (muli 2 height) quote,
+        "</tr>\n<tr>",
+            (foldl (lam str. lam x. concat str (makeTDElem x width height quote))) "" ["none","none","black","none","none"],
+        "</tr>\n</table>> "]
+        ,"") in
+    formatComponentToDot name quote (concatList ["label=",quote,quote,settings.0]) isConnected
+
+-- returns a custom component in dot.
+let circOtherToDot = lam quote. lam name. lam value. lam _. lam custom_settings. lam isConnected.
+    let settings = match custom_settings with Some (setting,unit) then (setting,unit) else
+        (foldl concat [] ["style=filled fillcolor=white shape=circle label=",quote,quote, " xlabel=",quote,value,quote]," ") in
+    let value_str = match value with "" then "" else (foldl concat [] [value," ",settings.1," "]) in
+    formatComponentToDot name quote (concatList ["xlabel=",quote,value_str,quote," ",settings.0]) isConnected
+
+-- returns a component in dot.
+let componentToDot = lam comp. lam quote. lam fig_settings.
+    match comp with Component (comp_type,name,maybe_value,isConnected) then
+        let figure_setting = 
+            let fig = find (lam x. setEqual eqchar x.0 comp_type) fig_settings in
```

Comment:
You should not use `setEqual eqchar` for string equality: this will only check that the two words contain the same letters, but not that they are in the same order. Use `eqstr` from `string.mc` instead. You have used `setEqual eqchar` in other places as well, they should also be fixed.

As a side note, this code shouldn't even run as you haven't imported `set.mc`. The only reason it does is that you have an unused import of `set.mc` in `circuit.mc`, which is imported before this file in `model.mc`.

Reply:
Fixed.